components/gas-estimator/GasEstimatorBox.tsx

```diff
@@ -0,0 +1,94 @@
+import { Box, BoxProps, Flex, Heading, Text } from "@chakra-ui/react";
+import { ContractType } from "@thirdweb-dev/sdk";
+import {
+  CONTRACT_TYPE_NAME_MAP,
+  GasEstimatorMap,
+  GasPrice,
+} from "constants/mappings";
+import { ethers } from "ethers";
+
+interface PriceLineProps {
+  gasPrice: number;
+}
+
+const PriceLine: React.FC<PriceLineProps> = ({ gasPrice, children }) => {
+  return (
+    <>
+      {gasPrice ? <Flex justifyContent="space-between">{children}</Flex> : null}
+    </>
+  );
+};
+
+interface GasEstimatorBoxProps extends BoxProps {
+  contractType: ContractType;
+  ethOrUsd: "eth" | "usd";
+  data: { gasPrice: number; ethPrice: number };
+}
+
+export const GasEstimatorBox: React.FC<GasEstimatorBoxProps> = ({
+  contractType,
+  ethOrUsd,
+  data,
+  ...props
+}) => {
+  const {
+    deployContract,
+    setClaimPhase,
+    batchUpload,
+    mint,
+    claim,
+    distributeFunds,
+  }: GasPrice = GasEstimatorMap[contractType];
+
+  const formatPrice = (price: number) => {
+    if (price && ethOrUsd === "eth") {
+      return `~${Number(
+        ethers.utils.formatUnits(
+          `${((data?.gasPrice as number) || 30) * price}`,
+          "gwei",
+        ),
+      ).toFixed(4)} ETH`;
+    } else if (price && ethOrUsd === "usd") {
+      return `~$${(
+        Number(
+          ethers.utils.formatUnits(
+            `${((data?.gasPrice as number) || 30) * price}`,
+            "gwei",
+          ),
+        ) * (data?.ethPrice || 3400)
```

Comment:
I don't think we should hard-code a price here if we can't get it. if we can't get the price just disable the toggle I think

Reply:
what about if we can't get the gasPrice?